src/PowerShellEditorServices/Services/PowerShell/Host/EditorServicesConsolePSHostRawUserInterface.cs

```diff
@@ -0,0 +1,322 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.PowerShell.EditorServices.Services.PowerShell.Console;
+using System;
+using System.Management.Automation;
+using System.Management.Automation.Host;
+using System.Threading;
+
+namespace Microsoft.PowerShell.EditorServices.Services.PowerShell.Host
+{
+    internal class EditorServicesConsolePSHostRawUserInterface : PSHostRawUserInterface
+    {
+        #region Private Fields
+
+        private readonly PSHostRawUserInterface _internalRawUI;
+        private readonly ILogger _logger;
+        private KeyInfo? _lastKeyDown;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Creates a new instance of the TerminalPSHostRawUserInterface
+        /// class with the given IConsoleHost implementation.
+        /// </summary>
+        /// <param name="logger">The ILogger implementation to use for this instance.</param>
+        /// <param name="internalHost">The InternalHost instance from the origin runspace.</param>
+        public EditorServicesConsolePSHostRawUserInterface(
+            ILoggerFactory loggerFactory,
+            PSHostRawUserInterface internalRawUI)
+        {
+            _logger = loggerFactory.CreateLogger<EditorServicesConsolePSHostRawUserInterface>();
+            _internalRawUI = internalRawUI;
+        }
+
+        #endregion
+
+        #region PSHostRawUserInterface Implementation
+
+        /// <summary>
+        /// Gets or sets the background color of the console.
+        /// </summary>
+        public override ConsoleColor BackgroundColor
+        {
+            get { return System.Console.BackgroundColor; }
+            set { System.Console.BackgroundColor = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the foreground color of the console.
+        /// </summary>
+        public override ConsoleColor ForegroundColor
+        {
+            get { return System.Console.ForegroundColor; }
+            set { System.Console.ForegroundColor = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the size of the console buffer.
+        /// </summary>
+        public override Size BufferSize
+        {
+            get => _internalRawUI.BufferSize;
+            set => _internalRawUI.BufferSize = value;
```

Comment:
Heads up @SeeminglyScience fixed a bug around this in the current code, may want to double check.

Reply:
It was `LengthInBufferCells` https://github.com/PowerShell/PowerShellEditorServices/pull/1571/files

Reply:
Oops, we didn't get this fixed. #1603